src/pybind11/PyBindImath/PyBindImathQuat.cpp

```diff
@@ -0,0 +1,205 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright Contributors to the OpenEXR Project.
+//
+
+#include "PyBindImath.h"
+
+#include <ImathVec.h>
+#include <ImathMatrixAlgo.h>
+#include <ImathEuler.h>
+
+namespace PyBindImath {
+
+namespace py = pybind11;
+
+using IMATH_NAMESPACE::Quat;
+using IMATH_NAMESPACE::Vec3;
+using IMATH_NAMESPACE::Matrix33;
+using IMATH_NAMESPACE::Matrix44;
+using IMATH_NAMESPACE::Euler;
+
+using IMATH_NAMESPACE::Quatf;
+using IMATH_NAMESPACE::Quatd;
+
+namespace {
+
+template<typename T>
+struct GetClassName {};
+
+template <>
+struct GetClassName<Quatf> {
+    static constexpr char const* value = "Quatf";
+};
+
+template <>
+struct GetClassName<Quatd> {
+    static constexpr char const* value = "Quatd";
+};
+
+template <typename Q>
+std::string Quat_str(Q const& v)
+{
+    std::stringstream stream;
+    stream << GetClassName<Q>::value << '(' 
+        << v[0] << ", " 
+        << v[1] << ", " 
+        << v[2] << ", " 
+        << v[3] << ')';
+    return stream.str();
+}
+
+template <typename Q, typename T = typename Q::value_type>
+std::string Quat_repr(Q const& v)
+{
+    static_assert(std::is_floating_point<T>::value, "Quat_repr requires a floating-point type");
+
+    // Set precision based on the type
+    constexpr int precision = std::is_same<T, float>::value ? 9 : 17;
+
+    std::ostringstream oss;
+    oss << GetClassName<Q>::value << '('
+        << std::setprecision(precision) << v[0] << ", "
+        << std::setprecision(precision) << v[1] << ", "
+        << std::setprecision(precision) << v[2] << ", "
+        << std::setprecision(precision) << v[3] << ')';
+    return oss.str();
+}
+
+} // namespace
+
+template <class Q, typename T = typename Q::value_type, typename E = Euler<T>>
+void register_Quat(py::module& m)
+{
+    auto className = GetClassName<Q>::value;
+
+    py::class_<Q>(m, className)
```

Comment:
We need integration unit tests for this feature

Reply:
There are comprehensive tests for the entire library in `src/test/PyImathtest/pyImathTest.in`. I hadn't figured out a strategy for bringing those test to the PyBind re-implementation as we build it up piece by piece, but ultimately, they are the reference.

Reply:
Oh I was unaware, I will have a look there! Thanks for the pointers!